File: src/frontend/src/app/core/services/barema.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { Barema, CreateBarema, UpdateBarema, FinalizarBarema } from '../models';

@Injectable({
  providedIn: 'root'
})
export class BaremaService {
  private endpoint = '/baremas';

  constructor(private api: ApiService) {}

  getAll() {
    return this.api.get<Barema[]>(this.endpoint);
  }

  getById(id: number) {
    return this.api.get<Barema>(`${this.endpoint}/${id}`);
  }

  getByCandidatoId(candidatoId: number) {
    return this.api.get<Barema[]>(`${this.endpoint}/candidato/${candidatoId}`);
  }

  getByAvaliadorId(avaliadorId: number) {
    return this.api.get<Barema[]>(`${this.endpoint}/avaliador/${avaliadorId}`);
  }

  create(data: CreateBarema) {
    return this.api.post<Barema>(this.endpoint, data);
  }

  updateCriterios(id: number, data: UpdateBarema) {
    return this.api.put<Barema>(`${this.endpoint}/${id}/criterios`, data);
  }

  finalizar(id: number, data: FinalizarBarema) {
    return this.api.post<Barema>(`${this.endpoint}/${id}/finalizar`, data);
  }

  delete(id: number) {
    return this.api.delete<void>(`${this.endpoint}/${id}`);
  }
}

File: src/frontend/ProcessoSelecao.Blazor/Services/BaremaService.cs
using ProcessoSelecao.Blazor.Models;

namespace ProcessoSelecao.Blazor.Services;

public class BaremaService
{
    private readonly ApiService _api;
    private const string Endpoint = "baremas";

    public BaremaService(ApiService api)
    {
        _api = api;
    }

    public virtual async Task<List<Barema>> GetAllAsync()
        => await _api.GetAsync<List<Barema>>(Endpoint) ?? new();

    public virtual async Task<Barema?> GetByIdAsync(int id)
        => await _api.GetAsync<Barema>($"{Endpoint}/{id}");

    public virtual async Task<List<Barema>> GetByCandidatoIdAsync(int candidatoId)
        => await _api.GetAsync<List<Barema>>($"{Endpoint}/candidato/{candidatoId}") ?? new();

    public virtual async Task<List<Barema>> GetByAvaliadorIdAsync(int avaliadorId)
        => await _api.GetAsync<List<Barema>>($"{Endpoint}/avaliador/{avaliadorId}") ?? new();

    public virtual async Task<Barema?> CreateAsync(CreateBarema dto)
        => await _api.PostAsync<Barema>(Endpoint, dto);

    public virtual async Task<Barema?> UpdateCriteriosAsync(int id, UpdateBarema dto)
        => await _api.PutAsync<Barema>($"{Endpoint}/{id}/criterios", dto);

    public virtual async Task<Barema?> FinalizarAsync(int id, FinalizarBarema dto)
        => await _api.PostAsync<Barema>($"{Endpoint}/{id}/finalizar", dto);

    public virtual async Task DeleteAsync(int id)
        => await _api.DeleteAsync($"{Endpoint}/{id}");

    public virtual async Task<List<ProgressoCandidato>> GetProgressoAsync(int processoId)
        => await _api.GetAsync<List<ProgressoCandidato>>($"{Endpoint}/progresso/{processoId}") ?? new();
}
